packages/next/src/lib/error-telemetry-utils.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Augments the digest field of errors thrown in React Server Components (RSC) with an error code.
+ * Since RSC errors can only be serialized through the digest field, this provides a way to include
+ * an additional error code that can be extracted client-side via `extractNextErrorCode`.
+ *
+ * The error code is appended to the digest string with a semicolon separator, allowing it to be
+ * parsed out later while preserving the original digest value.
+ */
+export const appendErrorCodeToDigest = (
+  thrownValue: unknown,
+  originalDigest: string
+): string => {
+  if (
+    typeof thrownValue === 'object' &&
+    thrownValue !== null &&
+    '__NEXT_ERROR_CODE' in thrownValue
+  ) {
+    return `${originalDigest};${thrownValue.__NEXT_ERROR_CODE}`
```

Comment:
should we use a separate sign rather than `;`, as `;` is a currently separator for some special digest like redirect errors. `${typeof REDIRECT_ERROR_CODE};${RedirectType};${string};${RedirectStatusCode};`

We might get confused or process it accidentally, maybe `@` or `#`